In [1]:
#Importing libraries 
%matplotlib inline
import statsmodels.api as sm
import numpy as np 
import scipy
import plotly.express as px
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns
from sklearn.impute import SimpleImputer
import warnings
warnings.filterwarnings(action="ignore")
import seaborn as sns



In [3]:
import sys, os
sys.path.append(os.path.abspath(os.path.join('../scripts')))

from data_preprocess import Data_preprocess
from data_description import Df_description
from plots import Plots


In [4]:
df = pd.read_csv('../data/ABtwoCampaignEngView.csv')

In [ ]:
df.shape

(912712, 10)

In [5]:
df.head()

Unnamed: 0                            auction_id experiment  \
0     1333639  8ec30aff-2332-4a1f-9330-b93abb32bc94    control   
1      730667  4e1bcb56-357c-4186-9d39-3be82e3519f1    control   
2     2069787  dd6ec327-fcfd-4a80-89c0-dccd3dd80f09    control   
3     1692366  b5222d0f-39e3-4c02-a12f-1ef09d4f861f    control   
4     1384123  9428377d-1504-4407-87c2-ce518f67eb87    control   

                                                date  hour device_make  \
0  <built-in method date of Timestamp object at 0...    15          LG   
1  <built-in method date of Timestamp object at 0...     7     Samsung   
2  <built-in method date of Timestamp object at 0...    14       Apple   
3  <built-in method date of Timestamp object at 0...    22       Apple   
4  <built-in method date of Timestamp object at 0...    21     Samsung   

   platform_os  browser  yes   no  
0          156        6  0.0  0.0  
1          162        6  0.0  0.0  
2          178       15  0.0  0.0  
3          167       15  0.0  0.0  
4          155       15  0.0  0.0

In [6]:
df.drop('Unnamed: 0', inplace=True, axis=1)
df.head()

auction_id experiment  \
0  8ec30aff-2332-4a1f-9330-b93abb32bc94    control   
1  4e1bcb56-357c-4186-9d39-3be82e3519f1    control   
2  dd6ec327-fcfd-4a80-89c0-dccd3dd80f09    control   
3  b5222d0f-39e3-4c02-a12f-1ef09d4f861f    control   
4  9428377d-1504-4407-87c2-ce518f67eb87    control   

                                                date  hour device_make  \
0  <built-in method date of Timestamp object at 0...    15          LG   
1  <built-in method date of Timestamp object at 0...     7     Samsung   
2  <built-in method date of Timestamp object at 0...    14       Apple   
3  <built-in method date of Timestamp object at 0...    22       Apple   
4  <built-in method date of Timestamp object at 0...    21     Samsung   

   platform_os  browser  yes   no  
0          156        6  0.0  0.0  
1          162        6  0.0  0.0  
2          178       15  0.0  0.0  
3          167       15  0.0  0.0  
4          155       15  0.0  0.0

# Data Cleaning

In [7]:
data_preprocess = Data_preprocess(df)
df_info = Df_description(df)

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 912712 entries, 0 to 912711
Data columns (total 9 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   auction_id   912712 non-null  object 
 1   experiment   912712 non-null  object 
 2   date         912712 non-null  object 
 3   hour         912712 non-null  int64  
 4   device_make  912712 non-null  object 
 5   platform_os  912712 non-null  int64  
 6   browser      912712 non-null  int64  
 7   yes          912712 non-null  float64
 8   no           912712 non-null  float64
dtypes: float64(2), int64(3), object(4)
memory usage: 62.7+ MB


In [9]:
# getting missing value column with percentage
df_info.find_column_based_missing_percentage()

total_missing_values missing_percentage
auction_id                      0              0.0 %
experiment                      0              0.0 %
date                            0              0.0 %
hour                            0              0.0 %
device_make                     0              0.0 %
platform_os                     0              0.0 %
browser                         0              0.0 %
yes                             0              0.0 %
no                              0              0.0 %

In [10]:
# finding duplicate rows
df_info.find_duplicates()

Empty DataFrame
Columns: [auction_id, experiment, date, hour, device_make, platform_os, browser, yes, no]
Index: []

In [11]:
# find unique value counts on columns
df_info.find_dataframe_columns_unique_value_count()

Unique Value Count
experiment                    2
yes                           2
no                            2
browser                       9
hour                         24
platform_os                  68
device_make                 255
auction_id               892951
date                     912712

There are only 892951 than the data's data point 912712.
* this means some participants had responded to the add multiple times.


In [12]:
# correlation among columns
df_info.find_matrix_correlation()

hour  platform_os   browser       yes        no
hour         1.000000    -0.010857  0.010845 -0.030519  0.005676
platform_os -0.010857     1.000000 -0.030597 -0.124713 -0.014569
browser      0.010845    -0.030597  1.000000 -0.039816  0.026515
yes         -0.030519    -0.124713 -0.039816  1.000000 -0.047435
no           0.005676    -0.014569  0.026515 -0.047435  1.000000